lib/boombox/pipeline.ex

```diff
@@ -10,11 +10,20 @@ defmodule Boombox.Pipeline do
         output: opts.output
       })
 
-    {[spec: spec], %{}}
+    {[spec: spec], %{parent: opts.parent}}
   end
 
   @impl true
+  def handle_child_notification(:external_resource_ready, _element, _context, state) do
+    send(state.parent, :external_resource_ready)
+    {[], state}
+  end
+
   def handle_child_notification(:processing_finished, :boombox, _ctx, state) do
     {[terminate: :normal], state}
   end
+
+  def handle_child_notification(_notification, _element, _context, state) do
+    {[], state}
+  end
```

Comment:
I am not sure we want to add the clause, I don't think we had it previously and we might miss some of the incoming messages by mistake.